kafka/protocol/types.py

```diff
@@ -100,7 +100,7 @@ def decode(cls, data):
 
     @classmethod
     def repr(cls, value):
-        return repr(value[:100] + b'...' if len(value) > 100 else b'')
+        return repr('<None>' if value is None else value[:100] + b'...' if len(value) > 100 else b'')
```

Comment:
You don't need to cast `None` to a string, `repr()` will do it automatically and it'll be less confusing (at least to me) to see the naturally expected string representation of `None`. Also, if you prepend the None check to the existing length check, then that will get evaluated first, preventing the `len()` from throwing the exception.

I also think there's a bug here and the final `b''` should actually be `value` as the whole point is to print the entire thing if it's less than 100 characters. But IIRC @tvoinarovskyi wrote this and he normally knows what he's doing, so perhaps there's a reason for this?

So all that put together probably looks like:

    value[:100] + b'...' if value is not None and len(value) > 100 else value

Reply:
Agreed on all counts. I'll wait for @tvoinarovskyi response for the final change in regards to the `b''` case.